### DISCUSSION TITLE
Hedgedoc for internal shared Q&A

### DISCUSSION THREAD

**Discussion by metazool** [NERC-CEH/rse_group#45] [Ideas]
The CodeRefinery workshops are making really nice use of Hedgedoc (a FOSS alternative to Google Docs, Etherpad etc) to run shared Q&A during their workshop sessions. We also saw this model (though not with Hedgedoc) used to great effect during some of the RSECon workshops.

There's a lot of potential in it both for ease-of-access evolving Q&A documents on technical topics, and a shared notespace during workshops and forums. We could also move the weekly meeting summary note-taking from within Sharepoint to Hedgedoc. 

What's the easiest path to setting up a Hedgedoc instance internal to the organisation? 

* [Helm charts for Hedgedoc](https://truecharts.org/charts/stable/hedgedoc/)
* [Hedgedoc overview](https://hedgedoc.org/)
* [Configuring authentication](https://docs.hedgedoc.org/configuration/#login-methods) if needed - easiest would be to keep it globally read/write but only accessible via the internal network / VPN

**Comment by metazool** [comment]
@iwalmsley @rodscott I'm not sure any of us have accounts on the EDS-maintained kubernetes cluster - is there bootstrap documentation about how to request access and to deploy onto it? If there isn't, we could use Hedgedoc to write some :D

**Comment by longr** [comment]
I have missed etherpad, and long thought it would be useful to use instead of docs/sharepoint for weekly/daily meetings.

**Comment by rodscott** [comment]
How does this compare to the [wiki](https://wiki.ceh.ac.uk/)? Just thinking in terms of what we already have as infrastructure. Could have a space set up for the RSE group.

**Reply by longr** [reply]
Thanks, that had missed me some how.  I think the main thing is that HedgeDoc is more like a google doc but markdown, so multiple people editing at once, where as the wiki is just a shared document.

**Reply by metazool** [reply]
Realtime collaborative editing, without the potential version conflicts inherent in updating a wiki page.

Ease of access for researchers without having to register for another service - if this were group or project internal documentation the wiki might be suitable, but this is explicitly pitched as a low-barrier outreach and collaboration tool

**Comment by iwalmsley** [comment]
We're really trying to rationalize/streamline our infrastructure and my concern is that this would be another thing to maintain. For this reason I'd probably still point to OneDrive (which I think has real time collab?).

**Reply by metazool** [reply]
I appreciate the concern about wanting to get a real handle on an asset registry and evolve EDS out of a situation where a volume of trains-on-time maintenance work displaces the opportunity to do more creative research infrastructure devops work! 

It's also important to make choices about research collaboration tools that are on a sustainable path, and ideally an open one. The suggestion of Hedgedoc doesn't come arbitrarily - it's also recommended by [NCSA DecentCI](https://decentci.ncsa.illinois.edu/) in their stack of "open source tools and services to meet the essential needs of most scientific research collaborations".

The Microsoft suite may provide "good enough" versions of a lot of collaboration infrastructure, but I (personally) want to push back against the tendency to compound into it - each move in that direction makes it twice as hard to move away. You must know in your hearts that Sharepoint / OneDrive are joyless, sterile environments to work in, parts of a one-size-fits-all template imposed upon research and education culture, not drawing from it. Research technology invention, trying to offer delightful interfaces, it's part of all our roles at EDS!

**Reply by iwalmsley** [reply]
I don't disagree, and am with you in general on embracing open source software. 

I would simply say that hosting any tools like this shouldn't (and maybe should never have been) part of our core competency within EDS. This could be something that the IT group could deploy and maintain - as it certainly could bring value to the entire org.

**Reply by metazool** [reply]
Thank you @iwalmsley - I haven't forgotten about this thread!

Where's the right forum to connect with the part of the IT group that maintains a k8s cluster and a register of applications on it?

What would we be advised to offer by way of a walkthrough for setup and configuration?

**Reply by iwalmsley** [reply]
I don't think IT currently have a k8s cluster which can be used by other teams in a multi-tenancy fashion - but it is planned for the short/medium term. It's definitely something we'll be looking to move to once it's ready.

Maybe we can chat (with @rodscott and @longr) about what the options could be using our infrastructure if you're willing to support it yourself.

**Reply by metazool** [reply]
That sounds good to me - view it as supporting the service for a trial period, documenting any benefits as well as its configuration. Unlikely as this sounds, I'd be glad to have a practical reason to put a foot into the world of maintaining asset registers, planning for support arrangements and communicating about infrastructure changes!

I'm on a longish leave soon, it makes sense to pick this up at the start of November...
